src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -87,9 +87,6 @@ def __init__(self, p, g, q=None):
         if q is not None and not isinstance(q, six.integer_types):
             raise TypeError("q must be integer or None")
 
-        if q is None and g not in (2, 5):
-            raise ValueError("DH generator must be 2 or 5")
```

Comment:
Actually once these two lines are removed it works fine. Indeed Scapy uses instances of DHParameterNumbers, but this never calls DH_check.

As far as I can tell, for now DH_check is called only in asymmetric/dh.py for `DHPrivateNumbers` through `load_dh_private_numbers` (or in test_dh.py through `dh_parameters_supported`). So apart from the library tests it is not called when dealing with `DHParameterNumbers` or `DHPublicNumbers`.